### PR TITLE
Always use private security group

### PR DESCRIPTION
**Pull request by tthyer**
- always use private security group
- if public network selected, additionally attach public security group

### CODE REVIEW

File: ec2/sc-ec2-linux-jumpcloud.yaml [hunk 1]
```diff
@@ -235,7 +235,6 @@ Resources:
         - !Ref 'InstanceRole'
   PrivateSubnetSecurityGroup:
     Type: 'AWS::EC2::SecurityGroup'
-    Condition: UsePrivateSubnet
```
Comment (zaro0508) [inline ec2/sc-ec2-linux-jumpcloud.yaml]: removing will always create this resource even if it's not being used?

Comment (tthyer) [reply ec2/sc-ec2-linux-jumpcloud.yaml]: It will always be used, as with the templates in aws-infra

Comment (zaro0508) [reply ec2/sc-ec2-linux-jumpcloud.yaml]: actually all ports are open for private instances so you don't even need a PrivateSecurityGroup.  The security group in managed-ec2-linux-v1.j2 is only created and used when the user specifies OpenPorts and it only applies to public instances

File: ec2/sc-ec2-linux-jumpcloud.yaml [hunk 2]
```diff
@@ -436,15 +435,15 @@ Resources:
...
-          'Fn::Sub': '${AWS::Region}-internalpoolvpc-PrivateSubnet'
+        - UsePublicSubnet
         - !ImportValue
           'Fn::Sub': '${AWS::Region}-internalpoolvpc-PublicSubnet'
+        - !ImportValue
+          'Fn::Sub': '${AWS::Region}-internalpoolvpc-PrivateSubnet'
       SecurityGroupIds: !If
-        - UsePrivateSubnet
+        - UsePublicSubnet
+        - [!Ref PrivateSubnetSecurityGroup, !Ref PublicSubnetSecurityGroup]
```
Comment (zaro0508) [inline ec2/sc-ec2-linux-jumpcloud.yaml]: why are both private subnet and public subnet here?

File: ec2/sc-ec2-linux-jumpcloud.yaml (hunk 1 above)
Comment (zaro0508) [reply ec2/sc-ec2-linux-jumpcloud.yaml]: sorry i misunderstood that your PrivateSubnetSecurityGroup is actually just the same as the Vpn security group.  this looks good.

File: ec2/sc-ec2-linux-jumpcloud.yaml [hunk 3]
```diff
@@ -436,15 +435,15 @@ Resources:
...
         - [!Ref PrivateSubnetSecurityGroup]
```
Comment (zaro0508) [inline ec2/sc-ec2-linux-jumpcloud.yaml]: why not just do 
```
- !ImportValue
              'Fn::Sub': '${AWS::Region}-${VpcName}-VpnSecurityGroup
```
like https://github.com/Sage-Bionetworks/aws-infra/blob/master/templates/managed-ec2-ubuntu-v1.yaml#L507

Comment (zaro0508) [reply ec2/sc-ec2-linux-jumpcloud.yaml]: ohh yeah, reason is to avoid imports in libs

**Pull request merged**